python/ray/tune/integration/wandb.py

```diff
@@ -145,11 +146,12 @@ def run(self):
     def _handle_result(self, result):
         config_update = result.get("config", {}).copy()
         log = {}
+        flat_result = flatten_dict(result.copy(), delimiter="/")
 
-        for k, v in result.items():
-            if k in self._to_config:
+        for k, v in flat_result.items():
+            if any(k.startswith(item) for item in self._to_config):
```

Comment:
Hm, should we maybe check for `k.startswith(item + "/") or k == item`? The `startswith` might be too broad if someone uses `exclude=loss` but passes `results={"losses": {"val": ...}}` or so.

Reply:
`k == item` would not work if the user wants to exclude whole sub-dicts (such as `config`) since all keys are replaced with `config/*`. `k.startswith(item + "/")` on the other hand would not work if the key to remove is not a nested dict (and was therefore not flattened). What if we require the items in `_to_config` and `_exclude` regular expressions?

Reply:
Yeah, but the combined expression `k.startswith(item + "/") or k == item` would do exactly what we want, right? If it is a nested dict, all keys start with `config/` and would be excluded by the `startswith`. Unnested keys would be caught by `k == item`

Reply:
Just to be sure, I mean `or` as in `||`, not as in "either/or"

Reply:
Oh right, I didn't realize that the `or` was part of the expression! I agree, makes sense.